### ISSUE TITLE
Need to be able to change the callback URL in OAuth2 flow

### ISSUE THREAD

**Issue by tbarn** [stoplightio/desktop#266]
For OAuth2 flow type `accessCode`, also known as authorization code

Need to be able to change the callback URL: <img width="533" alt="screen shot 2018-07-25 at 1 39 21 pm" src="https://user-images.githubusercontent.com/2008034/43220666-35644172-9010-11e8-96a8-6ad3eeee8140.png">

> server rejects the token request because the real callback URL used when I click on Get Access token button is not the same

**Comment by agonzalezml** [comment]
The real problem is:
When you send the user to the authorization url, stoplight is setting on the redirect_uri query param an url that not correspond exactly with the callback url, stoplight is adding to the query of that url some extra params, example:
```
https://beta.leverade.com/oauth/authorize
    ?response_type=code
    &client_id=1
    &redirect_uri=https://next-api.stoplight.io/oauth_token_capture
        ?cp=https://next.stoplight.io/leverade/leverade/version/1.0/leverade.oas2.yml
            ?view=/user/show-current-user
            &client_id=1
            &client_secret=this is absolutely secret!
            &access_token_url=https://api.leverade.com/oauth/token
            &authorize_url=https://beta.leverade.com/oauth/authorize
    &scope=*
    &state=HJs6U9TNX
```
As defined in the OAuth specification, the redirect_uri must be exactly the callback url, and can't have extra query params. If you need keep some state you have the state param.
And also, **you can't expose the client_secret on the url**.
The url should be similar to the following:
```
https://beta.leverade.com/oauth/authorize
    ?response_type=code
    &client_id=1
    &redirect_uri=https://next-api.stoplight.io
    &scope=*
    &state=HJs6U9TNX
```

**Comment by lottamus** [comment]
This flow has been fixed and will go out in the 4.4 release.
